packages/feeds-client/src/event-handlers/feed/handle-follow-deleted.ts

```diff
@@ -0,0 +1,102 @@
+import type { Feed, FeedState } from '../../Feed';
+import type { FollowResponse } from '../../gen/models';
+import type {
+  EventPayload,
+  PartializeAllBut,
+  UpdateStateResult,
+} from '../../types-internal';
+import {
+  getStateUpdateQueueId,
+  shouldUpdateState,
+} from '../../state-updates/state-update-queue';
+import { isFeedResponse } from '../../type-assertions';
+
+export const updateStateFollowDeleted = (
```

Comment:
@szuperaz - I kept these as there are already tests you wrote for it but I'm going to create a different set of tests for the `handleFollowUpdated` - we can re-adjust if we don't agree on the approach (I'm going to be following the approach we chose for Thread class in chat).

On a side note; I just moved everything that was related together, I feel like this way it's easier to follow.